analyses/org.eclipse.tracecompass.incubator.lttng2.ust.extras.core/src/org/eclipse/tracecompass/incubator/internal/lttng2/ust/extras/core/pthread/PThreadLockGraphHandler.java

```diff
@@ -131,23 +132,26 @@ private void handleAcquireLockEvent(ITmfEvent event) {
         OsWorker worker = getOrCreateKernelWorker(event, tid);
 
         // Get the vertex for the last request
-        TmfVertex lastReqVertex = fLastRequest.get(tid, fieldValue);
+        ITmfVertex lastReqVertex = fLastRequest.get(tid, fieldValue);
         if (lastReqVertex == null) {
             return;
         }
 
         // Get the last lock owner
         LastLockOwner lastOwner = fLastLockOwner.get(fieldValue);
-        if (lastOwner != null && lastOwner.fTid != tid && lastOwner.fVertex.getTs() > lastReqVertex.getTs()) {
+        if (lastOwner != null && lastOwner.fTid != tid && lastOwner.fVertex.getTimestamp() > lastReqVertex.getTimestamp()) {
             // This thread has been blocked, add the proper vertices and links
-            TmfGraph graph = Objects.requireNonNull(fProvider.getAssignedGraph());
+            ITmfGraph graph = Objects.requireNonNull(fProvider.getGraph());
+
             // First add a vertex at the time of lock request
-            graph.append(worker, lastReqVertex, EdgeType.RUNNING);
+            graph.append(lastReqVertex, new OSEdgeContextState(OSEdgeContextEnum.RUNNING));
```

Comment:
OSEdgeConstextState is internal. Should it be public API or is there a different (more correct) way to create ITmfContextState here?

Reply:
Indeed there should. I would suggest that's a different patch, for now, we'll grin and bare it.